tests/acceptance/bootstrap/PublicWebDavContext.php

```diff
@@ -53,61 +53,53 @@ public function getPublicDavVersion(string $versionString): int {
 
 	/**
 	 * @param string $range ignore if empty
-	 * @param string $publicWebDAVAPIVersion
 	 * @param string|null $password
 	 *
 	 * @return ResponseInterface
 	 */
-	public function downloadPublicFileWithRange(string $range, string $publicWebDAVAPIVersion, ?string $password = ""):ResponseInterface {
-		if ($publicWebDAVAPIVersion === "new") {
-			// In this case a single file has been shared as a public link.
-			// Even if that file is somewhere down inside a folder(s), when
-			// accessing it as a public link using the "new" public webDAV API
-			// the client needs to provide the public link share token followed
-			// by just the name of the file - not the full path.
-			$fullPath = (string) $this->featureContext->getLastCreatedPublicShare()->path;
-			$fullPathParts = \explode("/", $fullPath);
-			$path = \end($fullPathParts);
-		} else {
-			$path = "";
-		}
+	public function downloadPublicFileWithRange(string $range, ?string $password = ""):ResponseInterface {
+		// In this case a single file has been shared as a public link.
+		// Even if that file is somewhere down inside a folder(s), when
+		// accessing it as a public link using the public webDAV API
+		// the client needs to provide the public link share token followed
+		// by just the name of the file - not the full path.
+		$fullPath = (string) $this->featureContext->getLastCreatedPublicShare()->path;
+		$fullPathParts = \explode("/", $fullPath);
+		$path = \end($fullPathParts);
+
 		return $this->downloadFileFromPublicFolder(
 			$path,
 			$password,
-			$range,
-			$publicWebDAVAPIVersion
+			$range
 		);
 	}
 
 	/**
-	 * @When /^the public downloads the last public link shared file using the (old|new) public WebDAV API$/
-	 * @When /^the public tries to download the last public link shared file using the (old|new) public WebDAV API$/
-	 *
-	 * @param string $publicWebDAVAPIVersion
+	 * @When /^the public downloads the last public link shared file using the public WebDAV API$/
+	 * @When /^the public tries to download the last public link shared file using the public WebDAV API$/
 	 *
 	 * @return void
 	 */
-	public function downloadPublicFile(string $publicWebDAVAPIVersion):void {
-		$response = $this->downloadPublicFileWithRange("", $publicWebDAVAPIVersion);
+	public function downloadPublicFile():void {
+		$response = $this->downloadPublicFileWithRange("", "");
```

Comment:
the parameter range still will be empty string here. Maybe I will look over the use of range and remove on next PR

Reply:
is there no default?

Reply:
```php
public function downloadPublicFileWithRange(string $range, ?string $password = "")
```

Reply:
if we are passing empty string then we can have it as default.

Reply:
Many function that uses this have second parameter `password`, so removing or making this optional don't change anything unless we interchange the parameter of downloadPublicFileWithRange so that password could be given and other parameter be optional

Reply:
hmm, I mean why can't we do this:
```php
public function downloadPublicFileWithRange(?string $range = "", ?string $password = "")
```
and use it like this:
```php
$this->downloadPublicFileWithRange()
// or
$this->downloadPublicFileWithRange("", $password)
```